include/deal.II/base/tensor_product_polynomials.h

```diff
@@ -62,7 +63,7 @@ DEAL_II_NAMESPACE_OPEN
  * 2003
  */
 template <int dim, typename PolynomialType = Polynomials::Polynomial<double>>
-class TensorProductPolynomials
+class TensorProductPolynomials : ScalarPolynomialsBase<dim>
```

Comment:
You'll have to do the derivation as `public:`

Reply:
I just saw that on my end! I was thinking "what do you mean the base class is inaccessible?" :P